packages/x-components/src/plugins/x-plugin.ts

```diff
@@ -345,12 +345,8 @@ export class XPlugin implements PluginObject<XPluginOptions> {
     forEach(
       customizedStoreEmitters,
       (event, stateSelector: AnySimpleStateSelector | AnyStateSelector) => {
-        const {
-          selector,
-          immediate = false,
-          filter = () => true,
-          ...options
-        } = this.isSimpleSelector(stateSelector) ? { selector: stateSelector } : stateSelector;
+        const { selector, immediate, filter, ...options } =
+          this.normalizeStateSelector(stateSelector);
```

Comment:
This was throwing an eslint error. It was complaining that we can't use unbound functions. To be honest I didn't understand why it was throwing that errors, `AnyStateSelector` and `AnySimpleSelector` are defined using the arrow function syntax, which should prevent these errors from happening. 
However I found that extracting this logic to a normalise function made TS stop complaining, so that is what I did.